src/Microsoft.IdentityModel.LoggingExtensions/IdentityLoggerAdapter.cs

```diff
@@ -0,0 +1,116 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Abstractions;
+
+namespace Microsoft.IdentityModel.LoggingExtensions
+{
+    /// <summary>
+    /// The default implementation of <see cref="IIdentityLogger"/> that provides a wrapper around <see cref="ILogger"/> instance.
+    /// </summary>
+    public class IdentityLoggerAdapter : IIdentityLogger
+    {
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Instantiates <see cref="IdentityLoggerAdapter"/> using <paramref name="logger"/>.
+        /// </summary>
+        /// <param name="logger">An <see cref="ILogger"/> instance to which identity log messages are written.</param>
+
+#pragma warning disable CS3001 // Argument type is not CLS-compliant
```

Comment:
For consistency, let's use the GlobalSuppressions pattern and target it to just the types you need.

Reply:
I attempted to use the Global Suppressions pattern, but it isn't working for this particular type. 
Intellisense doesn't auto recommend suppressing the error by adding to the Gobal Suppressions file.
The Code Analyzer doesn't recognize it either when I add the error to the suppressions file manually.